### DISCUSSION TITLE
Upgrade to IDF v5

### DISCUSSION THREAD

**Discussion by CarlosDerSeher** [CarlosDerSeher/snapclient#57] [Show and tell]
I pushed an upgrade to IDF v5.1.1 to a separate branch. There are still some deprecated warnings which need to be resolved but most of the work should be done. Works fine on my Lyrat Board

Edit: @luar123, @DerPicknicker as I understand you were waiting for it

**Comment by DerPicknicker** [comment]
Oh yeah. That's great. How you're building it? With the same docker command just change the version of the container?

**Reply by CarlosDerSeher** [reply]
Ah OK, I am using Ubuntu and eclipse with espressif plugin.

I think you just have to change the version for docker build

**Reply by DerPicknicker** [reply]
I will try it later. And give you an update...

**Reply by DerPicknicker** [reply]
Is your IDF 5 branch using the default i2s driver or your custom one? If it uses the default one I can try to update the TAS5805M integration.

**Reply by CarlosDerSeher** [reply]
It uses the default IDF driver.

Edit: that was the main reason for the upgrade

**Reply by DerPicknicker** [reply]
Cool, I can try to update the TAS5805M integration.

**Comment by luar123** [comment]
That's awesome, thanks! Tested it with PCM5102A DAC and seems to work (including sample stuffing).

**Reply by CarlosDerSeher** [reply]
Great news. I am working on the sample stuffing approach actively because it seems the results are even better regarding sync. Still I am currently evaluating and a push will happen very soon

**Comment by escalator2015** [comment]
Hi, I dont know if this is the best place to put this but it can be useful for everybody: 

If you have clicks or static noise in your audio signal from time to time or your signal has short drops of level without no failure indication when using ESP32 with wifi activated try to reduce the wifi  TX power.
Over some value the wifi radio is polluting the circuitry on chip so the interference is very noticiable when you use audio.
The command to adjust or to get the actual TX power is:
```
       int8_t power; 
       esp_wifi_set_max_tx_power(desired_value); 
       // from 34 to 82, real power in dBm is: desired_value * 0.25dBm
       //so range is from 8.5dBm to 20.5dBm, the lower the cleaner
       esp_wifi_get_max_tx_power(&power);
       ESP_LOGI(TAG,"NEW RF Power:%d.\n",power);
```

**Reply by DerPicknicker** [reply]
Great suggestion - I use some boards with a proper power supply. I can't remember having that drops. Only APLL caused here some issues. But this will also been resolved soon.

**Reply by escalator2015** [reply]
One of the reasons for Espressif to remove the wireless part from incoming ESP32-P4 was this interference. At default max. TX power is very evident, mainly if you are using any ADC for analog audio input. 
The problem is that RF circuitry is very close and in the same substrate as the other elements of the mcu so there is radiation and jitter that cannot be avoided from outside except to take the power emitted down.

**Reply by CarlosDerSeher** [reply]
> One of the reasons for Espressif to remove the wireless part from incoming ESP32-P4 was this interference

What makes you think that? It's true, esp32 internal adc and dac are very susceptible to interference but if you are using external audio codecs and have a good pcb design there won't be any problems at all. I don't see problems with lyrat v4.3 and my own custom design doesn't have problems either

**Comment by escalator2015** [comment]
https://esp32.com/viewtopic.php?t=33417
Maybe is because I used esp32 for a very specific purpose.
I design dsp circuits with converters for Pro Audio industry that are using very high power audio amplifiers so residual noise is critical at those levels of amplification so , in comparison, lyrat v4.3 converter (ES8388) is too noisy, even without wifi active.

**Reply by CarlosDerSeher** [reply]
I agree ES8388 isn't the best codec you can get :)
But still, if you design the system well I don't see how you'll get interference from esp to codec. At least not on the digital side. I guess when you need very low noise then a radio transmitter near the high gain analog stage will always be a no go?!
